src/components/Dataset/DatasetPageHeader.js

```diff
@@ -7,68 +8,46 @@ import { DatasetProcessingError } from './DatasetProcessingError'
 import { DatasetReady } from './DatasetReady'
 import { DatasetRegenerate } from './DatasetRegenerate'
 
-const Block = ({ children }) => {
-  const { setResponsive } = useResponsive()
-
-  return (
-    <FixedContainer>
-      <Box
-        pad={{
-          vertical: setResponsive('basex6', 'basex8', 'basex14')
-        }}
-      >
-        {children}
-      </Box>
-    </FixedContainer>
-  )
-}
-
 export const DatasetPageHeader = ({ dataset }) => {
   const { setResponsive } = useResponsive()
-  const { isFailed, isProcessing, isReady, isReadyExpired } =
-    getDatasetState(dataset)
-
-  if (isProcessing) {
-    return (
-      <Block>
-        <DatasetProcessing dataset={dataset} />
-      </Block>
-    )
+  const [datasetStates, setDatasetStates] = useState({})
+
+  useEffect(() => {
+    setDatasetStates(getDatasetState(dataset))
+  }, [dataset])
+
+  // maps dataset states to their corresponding component
+  const statusHeaders = {
+    isProcessing: DatasetProcessing,
+    isFailed: DatasetProcessingError,
+    isReady: DatasetReady,
+    isReadyExpired: DatasetRegenerate
   }
 
-  if (isFailed) {
-    return (
-      <Block>
-        <DatasetProcessingError dataset={dataset} />
-      </Block>
-    )
-  }
-
-  if (isReady) {
-    return (
-      <Block>
-        <DatasetReady dataset={dataset} />
-      </Block>
-    )
-  }
-
-  if (isReadyExpired) {
-    return (
-      <Block>
-        <DatasetRegenerate dataset={dataset} />
-      </Block>
-    )
-  }
+  // finds the current dataset state
+  const currentState = Object.keys(statusHeaders).find(
+    (state) => datasetStates[state]
+  )
+  // gets the corresponding component for the current dataset state
+  const Component = currentState ? statusHeaders[currentState] : null
```

Comment:
```suggestion
  const Component = statusHeaders[currentState]
```

If currentState is evaluated on load then this can simply just access the key.

Reply:
It has been updated and with the addition of a new component `DatasetNotProcessed` for `isNotProcessed` state 👍